src/app/content/utils/canonicalUrl.ts

```diff
@@ -9,19 +9,22 @@ export async function getCanonicalUrlParams(
   archiveLoader: AppServices['archiveLoader'],
   osWebLoader: AppServices['osWebLoader'],
   bookId: string,
-  pageShortId: string
+  pageShortId: string,
+  bookVersion?: string
 ) {
   const getBook = makeUnifiedBookLoader(archiveLoader, osWebLoader);
+
   const canonicals = [
     ...(CANONICAL_MAP[bookId] || []),
-    bookId, // use the current book as a last resort
+    ...(bookVersion === BOOKS[bookId].defaultVersion ? [bookId] : []),
```

Comment:
does this throw if you load a book not in the books config?

Reply:
I think so? I wasn't sure when is the app supposed to work in production when the id is not in `BOOKS`. I've updated it.

Reply:
in production it shouldn't be possible to load a book outside the BOOKS config, but it seemed like this code would throw in development when loading dynamic books